tests/unit/test_retry.py

```diff
@@ -0,0 +1,164 @@
+import logging
+from datetime import datetime, timedelta
+from os import getenv
+from typing import AsyncIterator
+from unittest.mock import AsyncMock, MagicMock
+
+import pytest
+
+import jbi.runner
+from jbi.bugzilla.models import Bug, WebhookEvent, WebhookRequest
+from jbi.queue import QueueItem, get_dl_queue
+from jbi.retry import retry_failed
+
+RETRY_TIMEOUT_DAYS = getenv("RETRY_TIMEOUT_DAYS", 7)
+
+
+@pytest.fixture()
+def logger():
+    logger = logging.getLogger(__name__)
+    logger.info = MagicMock()
+    logger.warn = MagicMock()
+    logger.error = MagicMock()
+    logging.getLogger = lambda x=None: logger
+    return logger
+
+
+@pytest.fixture()
+def execute_action():
+    _execute_action = MagicMock()
+    jbi.runner.execute_action = _execute_action
+    return _execute_action
```

Comment:
Maybe you could add a parameter to retry_failed so it's 
```python
def retry_failed(item_executor=runner.execute_action)
```

So that it tests you can do:

```python
retry_failed(item_executor=MagicMock())
```

Reply:
I'll make that change. Is this the "python way" of doing things? I'm used to using a lot of dependency injection for tests in other languages, but was treating python more like the looser languages (like node) and mocking them directly.